pkg/arvo/lib/eth-abi.hoon

```diff
@@ -0,0 +1,559 @@
+=,  ethereum-types
+
+|%
++$  etyp  etyp:abi:ethereum
+
++$  abi-events  (pair (map @ux @tas) (map @tas (pair (list etyp) type)))
+
++$  contract
+  $:  name=@tas
+      write-functions=(map @tas function)
+      read-functions=(map @tas function)
+      events=(map @ux event)
+  ==
+
++$  function
+  $:  name=@tas
+      sol-name=@tas
```

Comment:
This is might have been a morally poor decision, but since solidity events are CamelCase, and functions are camelCase, I made them instead %hoon-case. A perhaps misguided act to enforce martian stylistic uniformity.